src/videouplinkbandwidthpolicy/NScaleVideoUplinkBandwidthPolicy.ts

```diff
@@ -99,4 +113,54 @@ export default class NScaleVideoUplinkBandwidthPolicy implements VideoUplinkBand
   setHasBandwidthPriority(hasBandwidthPriority: boolean): void {
     this.hasBandwidthPriority = hasBandwidthPriority;
   }
+
+  setTransceiverController(transceiverController: TransceiverController): void {
+    this.transceiverController = transceiverController;
+  }
+
+  async updateTransceiverController(): Promise<void> {
+    if (!this.transceiverController) {
+      return;
+    }
+    const encodingParams: RTCRtpEncodingParameters = {
+      scaleResolutionDownBy: this.scaleResolutionDownBy(),
+      maxBitrate: this.maxBandwidthKbps() * 1000,
+    };
+    if (
+      !this.isEncodingParameterEqual(
+        encodingParams,
+        this.encodingParamMap.get(NScaleVideoUplinkBandwidthPolicy.rid)
+      )
+    ) {
+      this.encodingParamMap.set(NScaleVideoUplinkBandwidthPolicy.rid, encodingParams);
+      this.transceiverController.setEncodingParameters(this.encodingParamMap);
+    }
```

Comment:
You're checking here for whether you need to do work at all.

Can you do so by checking your inputs (that is, same number of videos as last time and same max bitrate = no action needed)?

If we assume/require that `updateTransceiverController` implementations don't do work if they don't need it — that is, we guarantee that we will keep them up to date with whatever you picked, even if we reconnect etc — do we need additional checks prior to updating the sender?

Reply:
There is no inputs for this method right or are you referring that we should keep check of previous bit rate and scale down? Not sure how is that different that we just do a diff of the encoding parameter?
> If we assume/require that updateTransceiverController implementations don't do work if they don't need it — that is, we guarantee that we will keep them up to date with whatever you picked, even if we reconnect etc — do we need additional checks prior to updating the sender?

Could you elaborate? This is mainly to guard against constantly need from setting encoding parameters every time active speaker changes (right now we do this check in audio video controller).